sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachineScaleSet.java

```diff
@@ -1790,6 +1812,7 @@ WithSystemAssignedIdentityBasedAccessOrCreate withSystemAssignedIdentityBasedAcc
                 DefinitionStages.WithApplicationSecurityGroup,
                 DefinitionStages.WithSecrets,
                 DefinitionStages.WithPlan,
+                DefinitionStages.WithEphemeralOSDisk,
```

Comment:
No Need to support UpdateStages?

Reply:
No, ephemeral os disk can only be specified during vm or vmss creation.
In this [document](https://docs.microsoft.com/en-us/azure/virtual-machines/ephemeral-os-disks#frequently-asked-questions)'s Q&A session, there is:
> Q: Can the ephemeral OS disk be applied to existing VMs and scale sets?
    A: No, ephemeral OS disk can only be used during VM and scale set creation.

And no update options exist in swagger parameters.

Reply:
And placement of the ephemeral os disk can't be changed afterwards, though it says it can be changed through recreation using ARMTemplate...
> Q: Can the ephemeral OS disk placement be modified after creation of VM?
  A: No, once the ephemeral OS disk is provisioned, the OS disk placement cannot be changed. But the VM can be recreated via ARM template deployment/PowerShell/CLI by updating the OS disk placement of choosing. This would result in the recreation of the VM with Data on the OS disk deleted and OS is reprovisioned.

Reply:
I assume change on OS disk will not happen when VM running.
Maybe it could be changed after shutdown?

I assume the "recreation" means the createOrUpdate PUT API, not the update PATCH API.

Reply:
I see, I haven't tried the `PUT` yet, I'll give it a go. 
According to the doc:
>Ephemeral OS Disk
VMs and scale set instances cannot be stop-deallocated

I'll try when VM is running or stopped.

Reply:
I don't think you need to bother trying the `PUT`.

If user really want to do that, they can do a new `define`.

Reply:
ok